test/parallel/test-http-agent-maxsockets.js

```diff
@@ -0,0 +1,55 @@
+var common = require('../common');
+var assert = require('assert');
+var http = require('http');
+var Agent = require('_http_agent').Agent;
+
+var agent = new Agent({
+  keepAlive: true,
+  keepAliveMsecs: 1000,
+  maxSockets: 2,
+  maxFreeSockets: 2
+});
+
+var server = http.createServer(function (req, res) {
+  res.end('hello world');
+});
+
+function get(path, callback) {
+  return http.get({
+    host: 'localhost',
+    port: common.PORT,
+    agent: agent,
+    path: path
+  }, callback);
+}
+
+var count = 0;
+function done() {
+  if (++count !== 2) {
+    return;
+  }
+  var freepool = agent.freeSockets[Object.keys(agent.freeSockets)[0]];
+  assert.equal(freepool.length, 2,
+    'expect keep 2 free sockets, but got ' + freepool.length);
+  console.log('http keepalive agent maxsockets test success.');
```

Comment:
Can I suggest that you remove the console.log?  A passing test should preferably be a silent test.

Also, micro style nit: on the line above, the string should be indented by 2 more spaces (or you can just leave it out.)